command/cliconfig/config_windows.go

```diff
@@ -44,3 +47,24 @@ func homeDir() (string, error) {
 
 	return syscall.UTF16ToString(b), nil
 }
+
+func replaceFileAtomic(source, destination string) error {
+	// On Windows, renaming one file over another is not atomic and certain
+	// error conditions can result in having only the source file and nothing
+	// at the destination file. Instead, we need to call into the MoveFileEx
+	// Windows API function.
+	srcPtr, err := syscall.UTF16PtrFromString(source)
+	if err != nil {
+		return &os.LinkError{"replace", source, destination, err}
+	}
+	destPtr, err := syscall.UTF16PtrFromString(destination)
+	if err != nil {
+		return &os.LinkError{"replace", source, destination, err}
+	}
+
+	err := windows.MoveFileEx(srcPtr, destPtr, flags)
```

Comment:
you have a re-declared error here (windows file, so not tested directly)

Reply:
Good catch! Thanks.

Reply:
Okay... peeking at my local history, I see that I over-zealously simplified this function after I'd initially tested it, reorganizing this code so that it was all in one function rather than two, but then apparently neglected to actually test the resulting simplification. :confounded: 

With this code now split out into its own PR I don't have a manual testing codepath to run this against right now, so I've just verified that it's _buildable_ on Windows now and I'll retest it working on Windows once this is in and I've got `terraform login` running against it.